File: jonahshop/apps/shipping/serializers/address.py
from rest_framework import serializers
from oscar.core.loading import get_model


ShippingAddress = get_model("order", "ShippingAddress")


valDict = {'required': True, 'allow_blank': False}
class ShippingAddressSerializer(serializers.ModelSerializer):
    class Meta:
        model = ShippingAddress
        fields = [
            'first_name', 'last_name', 'phone_number', 'state',
            'line4', 'line1', 'country', 'postcode', 'notes'
        ]
        extra_kwargs = {
            'first_name': valDict,
            'last_name': valDict,
            'phone_number': valDict,
            'state': valDict,
            'line4': valDict,
            'line1': valDict,
        }

File: storefront/src/typedefs/address.ts
export interface IShippingAddress extends Record<string, any> {
  first_name: string,
  last_name: string,
  state: string,
  line4: string,
  line1: string,
  postcode: string,
  phone_number: string,
  country: string,
  notes: string,
}


export interface IUserAddress extends IShippingAddress {
  user: number 
  is_default_for_shipping: boolean
  is_default_for_billing: boolean
  num_orders_as_shipping_address: number
  num_orders_as_billing_address: number
  date_created: string
}